libs/category/src/filters/type/range/behaviors/apply/apply.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  DaffCategoryFilterRangeRequestBase,
+  DaffCategoryFilterRangeBase,
+} from '../../../../../models/public_api';
+import { daffCategoryComputeFilterRangePairLabel } from '../../compute-pair-label';
+
+/**
+ * Applies a filter request to a group of range type filter options.
+ *
+ * @idempotent
+ */
+export const daffApplyFilterRange = <T>(
+  request: DaffCategoryFilterRangeRequestBase<T>,
+  filter: DaffCategoryFilterRangeBase<T>,
+): DaffCategoryFilterRangeBase<T> => ({
+    ...filter,
+    options: {
+      [daffCategoryComputeFilterRangePairLabel(request.value.min, request.value.max)]: {
```

Comment:
Won't this lose idempotence? If there is a filter applied for `price: 30-40` and you add another filter for `price: 40-50`, shouldn't there be either two filters for `30-40` and `40-50` or one filter for `30-50`? This looks like it would override the original `price: 30-40`.

Reply:
We discussed this at length. We might support multiple range filters at some later date, but for now, applying a range filter option always removes the previous one.

Reply:
@lderrickable, I'm not sure this violates idempotence. 

Simply stated, idempotence implies that f(a) = b that f(b) = b. 

So, upon reapplying the function to a result, you will get the same result.

Since the range filters "apply" always gets you to the same result assuming you use the same request on the filters returned from the prior apply, apply is idempotent.